src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts

```diff
@@ -584,8 +586,14 @@ export class TableComponent implements AfterContentChecked, OnInit, OnChanges, O
     return _.isEmpty(css) ? undefined : css;
   }
 
-  ngOnChanges() {
-    this.useData();
+  ngOnChanges(changes: SimpleChanges) {
+    _.forIn(changes, (_change: SimpleChange, propName: string) => {
```

Comment:
I find the native `for-of` looping simpler:
```suggestion
    for (const propName in changes) {
```
Or even [simpler](https://dev.to/juanjoms/comment/10511):
```ts
ngOnChanges(changes: SimpleChanges) {
  if (changes.data && changes.data.currentValue) {
    this.userData();
  }
}
```